react/components/ProductKitContent.js

```diff
@@ -0,0 +1,94 @@
+import { equals } from 'ramda'
+import React, { Component, Fragment } from 'react'
+import { FormattedMessage } from 'react-intl'
+
+import { extractItemsKit } from '../helpers/ProductKitHelper'
+import ProductKitDetails from './ProductKitDetails'
+import ProductKitItem from './ProductKitItem'
+
+const MAX_VISIBLE_ITEMS = 3
+const DEFAULT_VISIBLE_ITEMS = Array(MAX_VISIBLE_ITEMS).fill(null)
+
+/**
+ * Product Kit component.
+ * Display a list of items which composes a kit.
+ */
+export default class ProductKitContent extends Component {
+  state = {
+    shownItems: DEFAULT_VISIBLE_ITEMS,
+    hidenItems: [],
+  }
+
+  componentDidMount() {
```

Comment:
I think that you could use getDerivedStateFromProps to handle this state changes

Reply:
The behavior of the componentDidUpdate and componentDidMount is different, in this case is not possible to use the getDerivedStateFromProps method since it is not possible to differ when the component has been updated or mounted.